modules/costs/app/contracts/time_entries/base_contract.rb

```diff
@@ -121,11 +124,11 @@ def validate_activity_active
     end
 
     def work_package_invisible?
-      model.work_package.nil? || !model.work_package.visible?(user)
+      model.entity.nil? || !model.entity.visible?(user)
     end
 
     def work_package_not_in_project?
-      model.work_package && model.project != model.work_package.project
+      model.entity && model.project != model.entity.project
```

Comment:
```suggestion
     project != entity.project
```

(this could also be inlined)